NinchatSDKSwift/Implementations/Helpers/NINLowLevelClientSession+Extension.swift

```diff
@@ -5,13 +5,16 @@
 //
 
 import Foundation
-import NinchatSDK
+import NinchatLowLevelClient
 
 extension NINLowLevelClientSession {
-    func send(_ param: NINLowLevelClientProps, _ payload: NINLowLevelClientPayload? = nil) throws -> Int {
-        var actionID: Int64 = 0
-        try self.send(param, payload: payload, actionId: &actionID)
+    func send(_ param: NINLowLevelClientProps, _ payload: NINLowLevelClientPayload? = nil) throws -> Int? {
+        var actionID = UnsafeMutablePointer<Int64>.allocate(capacity: 1)
+        try self.send(param, payload: payload, actionId: actionID)
         
-        return Int(actionID)
+        defer {
+            actionID.deallocate()
```

Comment:
Shouldn't defer immediately follow the allocation call?

Reply:
It doesn't matter where we put it, it is called once the function returns the actual value.

Reply:
What happens if send returns an error, and try propagates it? Surely the defer block never gets evaluated.

Reply:
So the code leaks memory?

Reply:
yes, I think so. But now we have replaced the code with the previous implementation that doesn't need explicit allocation and deallocation.